logstash-core/lib/logstash/config/config_ast.rb

```diff
@@ -76,6 +76,14 @@ def self.defered_conditionals_index=(val)
     @defered_conditionals_index = val
   end
 
+  def self.plugin_instance_count
+    @plugin_instance_count
```

Comment:
since this is pretty much the same logic as with `@defered_conditionals_index` I suggest we use the same naming? `@plugin_instance_index` ?